File: 876 - Middle of the Linked List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def middleNode(self, head: ListNode) -> ListNode:
        i, j = head, head
        c = 0
        while i is not None:
            i = i.next
            if c % 2 == 1:
                j = j.next
            c += 1
        return j

File: 876 - Middle of the Linked List.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var middleNode = function(head) {
    let i = head, j = head;
    let c = 0;
    while (i != null) {
        i = i.next;
        if (c++ % 2 == 1) {
            j = j.next;
        }
    }
    return j;
};